### PR TITLE
Tested bug #731 and bug 529

### PR DESCRIPTION
**Pull request by dankyalo599**
*Issue #, 731*

*Description of changes:*
for  bug #731 
I have written test cases to check the behavior of getTitle method for Email and SNS, which return the subject if it is not null or empty, otherwise the default title "Alerting-Notification Action" will be used.
for bug #529 
For slack, I have written test cases that the subject should be returned without any hardcoded prefix "Alerting-Notification Action" this will check for the bug described .

*CheckList:*
[ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/alerting/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: release-notes/opensearch-alerting.release-notes-2.5.0.0.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+## Version 2.5.0.0 2023-01-10
```
Comment (lezzago) [inline release-notes/opensearch-alerting.release-notes-2.5.0.0.md:1]: Why is this file added?

File: alerting/build.gradle [hunk 2]
```diff
@@ -80,6 +80,7 @@ configurations.testImplementation {
 }
 
 dependencies {
+    testImplementation 'org.junit.jupiter:junit-jupiter:5.8.1'
```
Comment (lezzago) [inline alerting/build.gradle:83]: Do we need to introduce a new dependency? Why cant we use the existing ones?

**Review by lezzago — CHANGES_REQUESTED** [review]
Please also remove the resources files and only include changes for your PR

**Comment by dblock** [comment]
Looks stalled, please feel free to finish and reopen @dankyalo599.

[Catch All Triage, attendees [1](https://github.com/dblock), [2](https://github.com/reta), [3](https://github.com/lukas-vlcek/), [4](https://github.com/rafaelma), [5](https://github.com/krisfreedain), [6](https://github.com/gargrajat), [7](https://github.com/andrross)]

**Pull request closed without merging**